Thermostat

Alloc186 opened the pull request:
Hi can you add also support for local control of thermostats? I mean climate entity class.

Thank you!

rospogrigio commented:
@postlund, I see you are still pushing into this PR which is inside mileperhour's repo... is this really your intention?

postlund commented:
Hmm, I don't really understand what is happening here and how can I have permissions to push here anyway....

postlund commented:
Uhm, this is a merge from master in your repo so it will keep updating every time something is merged to master in your fork. That's the problem.

pabigot committed:
Fix kelvin-to-mired conversion

Use the core utility rather than reimplementing it.

Signed-off-by: Peter A. Bigot <pab@pabigot.com>

pabigot committed:
Fix out-of-range color temperatures

If light.turn_on is invoked with `brightness: 128, color_temp: 500`
for a bulb that supports mired range 154..370 in value range 1..255
the calculated color temperature will be negative.  When the bulb
receives the command it may turn on, but will fail processing the
color_temp operation and will not send a response.

At this point `light.toggle` will have no effect because Home
Assistant is unaware that the bulb is on.

Fix the state by clamping the sent color temperature to the allowed
range.

Signed-off-by: Peter A. Bigot <pab@pabigot.com>

vmartinv committed:
use idle instead of off HVAC state

vmartinv committed:
add Manual/Program HVAC mode

rikman122 committed:
Merge branch 'master' into vacuum_platform_new

vmartinv committed:
add turn on/off method

rospogrigio committed:
Merge pull request #548 from vmartinv/climate

Implementation of climate platform

rikman122 committed:
Merge branch 'master' into vacuum_platform_new

Commit:
lint code style fix

rospogrigio committed:
Merge pull request #506 from rikman122/vacuum_platform_new

Vacuum platform new

Daandeve committed:
#711 Implemented m/p HVAC mode set

gpaesano committed:
Update climate.py

There is a typo in target precision which prevents corrct handling of temperature

mgax committed:
Improve the Readme

* Add link to _tinytuya_.
* Clarify steps in Home Assistant Configuration screens.
* Set `yaml` syntax on a code example.

alicialiali committed:
Add files via upload

alicialiali committed:
Rename pt_br.json to pt-BR.json

rospogrigio committed:
Update README.md

alicialiali committed:
Merge branch 'master' into master

rospogrigio committed:
Merge branch 'master' into patch-2

rospogrigio committed:
Merge pull request #722 from gpaesano/patch-2

Update climate.py

rospogrigio committed:
Merge branch 'master' into pabigot/20211031a

rospogrigio committed:
Merge pull request #611 from pabigot/pabigot/20211031a

Fix color temperature to use correct conversion and clamp to allowed range

alicialiali committed:
Merge branch 'master' into master

rospogrigio committed:
Merge pull request #743 from LeandroIssa/master

Add Brazilian Portuguese translation

rospogrigio committed:
Introduced TuyaCloudApi class

rospogrigio committed:
Introduced diagnostics

rospogrigio committed:
Modified config flow, introduced device deletion

rospogrigio committed:
Introduced migration; fixed update when adding device

rospogrigio committed:
Finalized device editing, device diagnostics, auto-update of local key and IP

rospogrigio committed:
Obfuscated diagnostics; fixed IP auto-update; cleaned logs

rospogrigio committed:
Obfuscated diagnostics; fixed IP auto-update; cleaned logs

rospogrigio committed:
Merge branch 'localtuya_4.0' of github.com:rospogrigio/localtuya-homeassistant into localtuya_4.0

rospogrigio committed:
Fixed device diagnostic corrupting config entry

rospogrigio committed:
First tox fixes

rospogrigio committed:
tox fixes

rospogrigio committed:
tox fixes

rospogrigio committed:
tox fixes

rospogrigio committed:
Removed local_key censoring in device diagnostics

rospogrigio committed:
Removed local_key censoring in device diagnostics

rospogrigio committed:
Removed local_key censoring in device diagnostics

rospogrigio committed:
Removed local_key censoring in device diagnostics

rospogrigio committed:
Removed local_key censoring in device diagnostics

rospogrigio committed:
Removed local_key censoring in device diagnostics

rospogrigio committed:
Added images for README

rospogrigio committed:
Added images for README

rospogrigio committed:
Adding images for README

rospogrigio committed:
Adding images for README

rospogrigio committed:
Update README.md

rospogrigio committed:
Update README.md

rospogrigio committed:
Update README.md

rospogrigio committed:
Add files via upload

rospogrigio committed:
Update README.md

rospogrigio committed:
Update info.md

rospogrigio committed:
Add files via upload

rospogrigio committed:
Add files via upload

rospogrigio committed:
Merge branch 'master' into localtuya_4.0

rospogrigio committed:
Fixed diagnostics currupting stored config_entry

rospogrigio committed:
Changed permissions for cloud_api.py

rospogrigio committed:
Fixed wrong 'already_configured' message when editing entities

sibowler committed:
Merge branch 'master' of https://github.com/rospogrigio/localtuya into rospogrigio-master

sibowler committed:
Merge branch 'rospogrigio-master'

sibowler committed:
Revert "Rebase from upstream"

sibowler committed:
Merge pull request #3 from sibowler/revert-2-master

Revert "Rebase from upstream"

sibowler committed:
Revert "Revert "Rebase from upstream""

sibowler committed:
Merge pull request #4 from sibowler/revert-3-revert-2-master

Revert "Revert "Rebase from upstream"" - fixes mistaken revert

rospogrigio committed:
Fixed Adding new device manually

rospogrigio committed:
Changed deprecated async_get_registry() to async_get()

rospogrigio committed:
Improvements on README

rospogrigio committed:
Fixed labels missing when editing Cloud configuration

rospogrigio committed:
Fix for 'KeyError: tuya_devices' at startup

rospogrigio committed:
Turned all 'print' messages into _LOGGER.debug

rospogrigio committed:
Improved code readability (racu10's suggestion)

wrt54g committed:
Components to integrations

rospogrigio committed:
Add files via upload

Updated cloud setup screenshot

rospogrigio committed:
Updated README for the 'Do not configure cloud API' button

rospogrigio committed:
Introduced checkbox to choose whether to setup Cloud API account

rospogrigio committed:
Merge branch 'master' into localtuya_4.0

rospogrigio committed:
Cleaned debug messages

rospogrigio committed:
Improved description of reload service; removed some print messages

rospogrigio committed:
Fixed restoring entity state when reloading

rospogrigio committed:
Merge pull request #837 from sibowler/master

Fix previous merge problem which removed entries for select entity from en.json

rospogrigio committed:
Merge branch 'master' into localtuya_4.0

elax46 committed:
Create it.json

Added Italian translation

rospogrigio committed:
Merge pull request #829 from rospogrigio/localtuya_4.0

Localtuya 4.0

rospogrigio committed:
Merge branch 'master' into url

rospogrigio committed:
Merge branch 'master' into improve-readme

rospogrigio committed:
Merge pull request #738 from mgax/improve-readme

Improve the Readme

alicialiali committed:
Add files via upload

elax46 committed:
Update it.json

Updated translation for Tuya 4.0

rospogrigio committed:
Update it.json

rospogrigio committed:
Update it.json

rospogrigio committed:
Optimized setup_entities by regrouping platforms

rospogrigio committed:
Update manifest.json

Commit:
Add "1/0" entry for HVAC_MODE_SETS

BHT-002 thermostat DP #4 set the heating mode (auto or manual). The accepted values are strings "0" for auto mode and "1" for manual mode.
This commit add an entry "1/0" to support this behaviour.

wrt54g committed:
Merge branch 'master' into url

deese committed:
Added DPS value type configuration

markvader committed:
Refactor number entity overrides

Relates to issues [929](https://github.com/rospogrigio/localtuya/issues/929), [939](https://github.com/rospogrigio/localtuya/issues/939) & [941](https://github.com/rospogrigio/localtuya/issues/941)
I do not have any suitable tuya devices in use to test against this number entity however the warning in logs "LocaltuyaNumber is overriding deprecated methods on an instance of NumberEntity] is now gone

https://developers.home-assistant.io/blog/2022/06/14/number_entity_refactoring/

sibowler committed:
Rebase from upstream

sibowler committed:
Fix NumberEntity HA changes and add step control

sibowler committed:
Rebase merge fix. Earlier commit also included changes for manually specifying DPS where entity detection failed, restoring DPS values on reconnect and also supporting devices which don't report DPS status until it has been set once.

sibowler committed:
Fixing formatting/style errors

sibowler committed:
Adding notes to README

sibowler committed:
Spelling correction

sibowler committed:
Updating documentation

sibowler committed:
Implementing RESET Tuya function to enable devices which are stuck in a zombie mode to start responding to commands. Added configurable DPS for Reset command as found devices wouldn't 'wake up' unless they had exactly the right entries

deese committed:
Merge branch 'master' into master

rospogrigio committed:
Update requirements_test.txt

rospogrigio committed:
Reverting requirements.txt

rospogrigio committed:
Merge branch 'master' into master

rospogrigio committed:
Release 4.0.2

rospogrigio committed:
Merge branch 'master' into master

sibowler committed:
Merge branch 'master' into restore_state

sibowler committed:
Fix for cover defect

sibowler committed:
Fix for cover issues with new state restore functionality

sibowler committed:
Fixing slight format error

rospogrigio committed:
Merge pull request #956 from sibowler/restore_state

Support Passive DPS devices, Restoring state after power off & NumberEntity enhancements

rospogrigio committed:
Edited manifest.json to 4.1.0

rospogrigio committed:
Ported README.md updates to info.md

deese committed:
Removed trailing space

deese committed:
Merge branch 'master' into master

sibowler committed:
Implementing safe default for devices not requiring RESET

sibowler committed:
Updating readme information to reflect the change.

sibowler committed:
Changes to make Passive DPS entites a configurable item

sibowler committed:
Updating documentation

deese committed:
Lint fix

rospogrigio committed:
Merge pull request #942 from deese/master

Added DPS value type configuration

rospogrigio committed:
Merge branch 'master' into fix_restart_behaviour

rsnodgrass committed:
added dict get() instead of assuming keys exist; now handles config_flow setup failures without crashing

rospogrigio committed:
Merge pull request #1022 from sibowler/fix_restart_behaviour

Implementing safe default for devices not requiring RESET/Passive functionality - Fix for bug in 4.1.0

rospogrigio committed:
Update manifest.json

rospogrigio committed:
Merge branch 'master' into url

rospogrigio committed:
Merge pull request #849 from wrt54g/url

Components to integrations

sibowler committed:
Fix for AttributeError: 'TuyaMessage' object has no attribute 'release' error

rospogrigio committed:
Merge pull request #712 from Daandeve/master

#711 Implemented m/p HVAC mode set

Picchios commented:
Hi rospogrigio,
would it be possible to integrate a thermostat of the manufacturer PLIKC, model "neve x w" ? It has the possibility to change to heat and cool and it would be nice to use it in local. Thank you.
Andrea

ov1d1u committed:
Allow calling localtuya.set_dp service by non-admin users

rospogrigio committed:
Introduced pytuya with support for 3.4 protocol

rospogrigio committed:
Introduced pytuya with support for 3.4 protocol

rospogrigio committed:
Introduced 3.4 protocol option in config flow

rospogrigio committed:
Introduced 3.4 protocol option in config flow

rospogrigio committed:
Fixed HEARTBEAT command

rospogrigio committed:
Fixed HEARTBEAT command

rospogrigio committed:
Added debugging

rospogrigio committed:
Added debugging

rospogrigio committed:
Fixed requested DPs for type_0d devices

rospogrigio committed:
Fixed requested DPs for type_0d devices

rospogrigio committed:
Fixed negotiation and sequence numbers for protocol 3.4

rospogrigio committed:
Fixed negotiation and sequence numbers for protocol 3.4

rospogrigio committed:
Fixed sequence numbering on status update

rospogrigio committed:
Fixed sequence numbering on status update

sibowler committed:
Merge branch 'rospogrigio:master' into fix_semaphore_release_error

rospogrigio committed:
Fixed tox issues

rospogrigio committed:
Fixed tox issues

rospogrigio committed:
Fixed tox issues

rospogrigio committed:
Fixed tox issues

rospogrigio committed:
Updated README.md and info.md

rospogrigio committed:
Updated README.md and info.md

rospogrigio committed:
Merge branch 'protocol_3.4' of github.com:rospogrigio/localtuya into protocol_3.4

rospogrigio committed:
Fixed tox issues

rospogrigio committed:
Fixed tox issues

rospogrigio committed:
Fix for version not being set yet when calling _setup_dispatcher

rospogrigio committed:
Fix for version not being set yet when calling _setup_dispatcher

rospogrigio committed:
Introduced abort in config flow if all entities are deselected when editing a device

rospogrigio committed:
Introduced the possibility to select which devices should have the pytuya debug enabled

rospogrigio committed:
Introduced device debugging option in README.md and info.md

rospogrigio committed:
Introduced IT and PT translations

rospogrigio committed:
Merge branch 'master' into device_debugging_option

rospogrigio committed:
Merge pull request #1227 from rospogrigio/device_debugging_option

Device debugging option

rospogrigio committed:
Update manifest.json for 5.0.0

rospogrigio committed:
New tuyadebug.tgz now supporting protocol 3.4

rospogrigio committed:
Update bug_report.md

rospogrigio committed:
Fix for unknown state of covers/shutters

rospogrigio committed:
New tuyadebug.tgz now including a tinytuya test

ov1d1u committed:
Merge branch 'rospogrigio:master' into master

sibowler committed:
Merge branch 'master' into fix_semaphore_release_error

CloCkWeRX committed:
Force to int

CloCkWeRX committed:
Adjust input to avoid putting a string into an int

CloCkWeRX committed:
Adjust input to avoid putting a string into an int

CloCkWeRX committed:
Swap to int

rospogrigio committed:
Improved logging

rospogrigio committed:
Improved stability and Fix local_key update to not be thwarted by retries

rospogrigio committed:
Merge branch 'sibowler-fix_semaphore_release_error' into stability_fixes

rospogrigio committed:
Merge branch 'sibowler-fix_semaphore_release_error' into stability_fixes

rospogrigio committed:
Merge branch 'CloCkWeRX-fix-casting-issue' into stability_fixes

rospogrigio committed:
Merge branch 'stability_fixes' of github.com:rospogrigio/localtuya-homeassistant into stability_fixes

rospogrigio committed:
Introduced update of local_key when editing a device

rospogrigio committed:
Merge branch 'stability_fixes' of github.com:rospogrigio/localtuya into stability_fixes

rospogrigio committed:
Merge branch 'stability_fixes' of github.com:rospogrigio/localtuya into stability_fixes

rospogrigio committed:
Improved logging

rospogrigio committed:
Merge branch 'stability_fixes' of github.com:rospogrigio/localtuya-homeassistant into stability_fixes

rospogrigio committed:
Tox fixes

rospogrigio committed:
Tox fixes

rospogrigio committed:
Tox fixes

rospogrigio committed:
Fix for not reconnecting after 'giving up'

rospogrigio committed:
Tox fixes

rospogrigio committed:
Tox fixes, reduced to 3.9 only

jlikeme committed:
Add entities during 'Edit a device'

jlikeme committed:
revert the default region

Commit:
Fix depreciated async_get_registry

Commit:
Merge pull request #1 from oven-lab/oven-lab-Fix-depreciated-async_get_registry

Fix depreciated async_get_registry

Commit:
Add HACS validation

Commit:
Remove extra key

Commit:
Update manifest.json

Commit:
Remove extra key

Commit:
Update hacs.json

Commit:
Add Hassfest validation

Commit:
Alphabetize manifest.json

Alphabetized manifest.json according to home assistant requirements.

Commit:
Delete combined.yaml

As described in KTibow/ha-blueprint, ha-blueprint is being retired and we should thereby switch over to hassfest and hacs validation.

Commit:
Update tox.yaml

Added workflow_dispatch

Commit:
Update tox.ini

rospogrigio committed:
Merge pull request #1400 from oven-lab/master

Code test changes and issue #1375, #1398

rospogrigio committed:
Merge branch 'master' into feat-add-entities

rospogrigio committed:
Reduced logging level for 'Command %d timed out waiting for sequence number'; set v.5.1.0

rospogrigio committed:
Merge branch 'master' into feat-add-entities

rospogrigio committed:
Fixed tox errors

rospogrigio committed:
Updated manifest.json

rospogrigio committed:
Fix for socket.send() exceptions in the logs

rospogrigio committed:
Fixing pylint error

rospogrigio committed:
Merge pull request #1424 from rospogrigio/fix_socket.send_exceptions

Fix for socket.send() exceptions in the logs

ov1d1u committed:
Merge branch 'rospogrigio:master' into master

Nealium committed:
Fixed initialization with API but no internet.

ncd7 committed:
fix bug when climate entity is using default min/max temp and temp unit is F

The default min/max temps are in C but when the climate entity is set up with F this will cause incorrect boundary check and an inability to change the temperature from the home assistant entity UI widget.

dulyts committed:
Handle 3.4 protocol, fix CI (#1251)

* Use refresh interval in version 3.4
* Scan interval convert to integer
* Try to fix CI
* Try to fix CI

rospogrigio committed:
Merge branch 'master' into offline_init

CodeFoodPixels committed:
Allow port reuse so that other integrations can read the same port (#1481)

Co-authored-by: rospogrigio <49229287+rospogrigio@users.noreply.github.com>

rospogrigio committed:
Merge branch 'master' into offline_init

rospogrigio committed:
Merge pull request #1465 from Nealium/offline_init

Fixes #1328

rospogrigio committed:
Fixing tox issues

rospogrigio committed:
Merge branch 'master' into master

rospogrigio committed:
Merge pull request #1135 from ov1d1u/master

Allow calling localtuya.set_dp service by non-admin users

rospogrigio committed:
Merge branch 'master' into climate-minmax-default-temp-when-in-F-fix

rospogrigio committed:
Merge pull request #1476 from ncd7/climate-minmax-default-temp-when-in-F-fix

fix bug when climate entity is using default min/max temp and temp un…

elad-bar committed:
Replaced SUPPORT_* with enums, set minimum HA version in HACS configuration to 2024.1.0

rospogrigio committed:
Merge pull request #1613 from elad-bar/master

Fix #1610 - Replaced SUPPORT_* with enums

simontegelid committed:
Add support for more ac/fan modes (#1389)

* Add support for more ac/fan modes

* Replace deprecated constants with enums

* Add support for ClimateEntityFeature.TURN_{ON,OFF}

* Fix swing log message

---------

Co-authored-by: Simon Tegelid <simon.tegelid@bitvis.io>

estebanz01 committed:
at init: Replace deprecated async entry setup call with suggestion.

Based on https://developers.home-assistant.io/blog/2024/06/12/async_forward_entry_setups/
it is possible to replace it with `async_forward_entry_setups` instead,
which accepts multiple platforms and it should be more efficient.

apbarratt committed:
Add discovered HVAC and PRESET options (#1177)

* Add discovered HVAC and PRESET options

This commit adds the HVAC Action Set and Preset Set that I've discovered my Bödenwarme Underfloor Heating thermostat to be using as detailed here: https://github.com/rospogrigio/localtuya/issues/1175

* Add HVAC mode and action sets for climate entities

These are the actions and modes as reported by the Magnum smart wifi
thermostat: https://www.magnumheating.com/product/remote-control/

---------

Co-authored-by: Rutger Kerkhoff <rutger.kerkhoff@grafana.com>

Dim145 committed:
fix: do not use deprecated features

Dim145 committed:
fix: do not use deprecated attr in light entity

Dim145 committed:
refactor: define MODE_MANUAL

Dim145 committed:
fix: add FanEntityFeature.TURN_OFF/ON to fans

Dim145 committed:
fix: add color_mode property & adapt color mode method (is_xxx_mode)

Dim145 committed:
fix: remove useless test line in status_updated of light

Dim145 committed:
fix: climate deprecated attributes

ialexyi committed:
add a subset of HVAC_MODE_SETS (#1875)

iPAC-40 Portable Air Conditioning Heat Pump with Wifi and Remote Control. These devices have a limited option of modes. The available modes are Hot, Cold and Dry.

Co-authored-by: rospogrigio <49229287+rospogrigio@users.noreply.github.com>

0Styless committed:
fix(common.py) - Change loglevel for missing device configuration from WARNING to DEBUG. (#1816)

Loglevel warning produces a huge amount of log messages (around 500000 a day - depends on how much tuya devices are in your local network).

Co-authored-by: Tobias Dörfler <doerfler.tobias@akdb.de>
Co-authored-by: rospogrigio <49229287+rospogrigio@users.noreply.github.com>

Handrail9 committed:
Update README.md (#1808)

Tuya integration page shows new information and no longer includes how to get Client ID & Secret. Added those here.

Co-authored-by: rospogrigio <49229287+rospogrigio@users.noreply.github.com>

rospogrigio committed:
Update manifest.json

Commit:
Deprecation fix: use async_register_admin_service

This fixes the following reported issue:

Detected that custom integration 'localtuya' accesses
hass.helpers.service, which should be updated to import functions used
from service directly at custom_components/localtuya/__init__.py,
line 165: hass.helpers.service.async_register_admin_service(. This will
stop working in Home Assistant 2025.5, please create a bug report at
https://github.com/rospogrigio/localtuya/issues

Commit:
Deprecation fix: do not call async_forward_entry_setups in an async task

This fixes the following reported issue:

Detected that custom integration 'localtuya' calls
async_forward_entry_setups for integration localtuya with title:
*** and entry_id: ***, during setup without awaiting
async_forward_entry_setups, which can cause the setup lock to be
released before the setup is done at
custom_components/localtuya/__init__.py, line 275:
await hass.config_entries.async_forward_entry_setups(entry, platforms).
This will stop working in Home Assistant 2025.1, please create a bug
report at https://github.com/rospogrigio/localtuya/issues

Commit:
Deprecation fix: use VacuumActivity

This fixes the following reported issue:

STATE_DOCKED was used from localtuya, this is a deprecated constant
which will be removed in HA Core 2026.1. Use VacuumActivity.DOCKED
instead, please report it to the author of the 'localtuya' custom
integration

STATE_ERROR was used from localtuya, this is a deprecated constant
which will be removed in HA Core 2026.1. Use VacuumActivity.ERROR
instead, please report it to the author of the 'localtuya' custom
integration

STATE_IDLE was used from localtuya, this is a deprecated constant
which will be removed in HA Core 2026.1. Use VacuumActivity.IDLE
instead, please report it to the author of the 'localtuya' custom
integration

STATE_PAUSED was used from localtuya, this is a deprecated constant
which will be removed in HA Core 2026.1. Use VacuumActivity.PAUSED
instead, please report it to the author of the 'localtuya' custom
integration

STATE_RETURNING was used from localtuya, this is a deprecated constant
which will be removed in HA Core 2026.1. Use VacuumActivity.RETURNING
instead, please report it to the author of the 'localtuya' custom
integration

rospogrigio committed:
Update config_flow.py

Fix for Config flow that could not be loaded after update to HA 2025.12 (Issue #2124)
